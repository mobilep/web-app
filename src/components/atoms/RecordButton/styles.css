.Button.RecordButton,
.Button.RecordButton:hover {
	height: 40px;
	background-color: transparent;
	color: var(--tomatoRed);
	outline: 0;
}

.RecordButton svg {
	width: 20px;
	height: 20px;
}

.RecordButton.recording svg {
	background-color: var(--tomatoRed);
	border-radius: 4px;
}
