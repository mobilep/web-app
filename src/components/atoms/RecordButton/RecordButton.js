import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { MessageBoxBaseButton } from '../..';
import { RecordIcon } from '../../../icons';
import './styles.css';

const getClassName = (className) => classNames('RecordButton', { [className]: className });

const RecordButton = ({ className, ...rest }) => (
	<MessageBoxBaseButton { ...rest } className={getClassName(className)}>
		<RecordIcon />
	</MessageBoxBaseButton>
);

RecordButton.propTypes = {
	className: PropTypes.string,
};

export default RecordButton;
