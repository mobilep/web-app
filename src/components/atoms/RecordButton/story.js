import React from 'react';
import { storiesOf } from '@storybook/react';

import RecordButton from './';

storiesOf('1. Atoms / RecordButton', module)
	.add('RecordButton', () => <RecordButton />);
