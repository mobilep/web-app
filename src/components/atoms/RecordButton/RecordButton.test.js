import React from 'react';
import { shallow } from 'enzyme';

import RecordButton from './RecordButton';

describe('Atom - RecordButton', () => {

	it('Should render', () => {
		const wrapper = shallow(<RecordButton />);
		expect(wrapper).toMatchSnapshot();
	});

});
