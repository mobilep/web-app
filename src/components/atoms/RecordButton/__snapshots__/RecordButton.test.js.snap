// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - RecordButton Should render 1`] = `
<MessageBoxBaseButton
  className="RecordButton"
>
  <RecordIcon />
</MessageBoxBaseButton>
`;
